sdk/js/src/Client.ts

```diff
@@ -200,10 +200,11 @@ export class DVCClient implements Client {
 
             let updatedUser: DVCPopulatedUser
 
+            const storedAnonymousId = this.store.loadAnonUserId()
             if (user.user_id === this.user.user_id) {
                 updatedUser = this.user.updateUser(user, this.options)
             } else {
-                updatedUser = new DVCPopulatedUser(user, this.options)
+                updatedUser = new DVCPopulatedUser(user, this.options, undefined, storedAnonymousId ?? undefined)
```

Comment:
is `storedAnonymousId ?? undefined` just handling the case that storedAnonymousId is null?

Reply:
yup, it is

Reply:
no need to block this PR for this, but this seems like something we should move to `loadAnonUserId`, so that it can just return `string | undefined` rather than `string | null | undefined`